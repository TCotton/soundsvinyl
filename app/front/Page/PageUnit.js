import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ExtendedPropTypes from 'extended-proptypes'
import {
	thumbnailOne,
	thumbnailTwo,
	thumbnailThree,
	thumbnailFour,
	thumbnailFive
} from '../../alternative_default_thumbnails'
import { Link } from 'react-router-dom'
import './pageUnit.scss'

export default class PageUnit extends Component {
	static propTypes = {
		id: ExtendedPropTypes.mongoId.isRequired,
		slug: PropTypes.string.isRequired,
		subtitle: PropTypes.string.isRequired,
		thumbnailUrl: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.func,
		]).isRequired,
		title: PropTypes.string.isRequired
	}

	constructor (props) {
		super(props)

		this.handleOnError = this.handleOnError.bind(this)
	}

	handleOnError (event) {
		const myArray = [
			thumbnailOne,
			thumbnailTwo,
			thumbnailThree,
			thumbnailFour,
			thumbnailFive
		]
		const thumbnailDefault = myArray[Math.floor(Math.random() * myArray.length)]
		const randomNumber = Math.floor(Math.random() * 8) + 1

		event.target.className = `random${randomNumber}`;
		event.target.src = thumbnailDefault;
		event.target.alt = '';
	}

	render () {
		const { slug, id, thumbnailUrl, title, subtitle } = this.props

		return (
			<Link
				styleName='pageUnit'
				to={`/${slug}/${id}`}
			>
				<img
					alt={title}
					onError={this.handleOnError}
					src={thumbnailUrl}
				/>
				<span className='video-title'>
					{title}
				</span>
				<span className='video-subtitle'>
					{subtitle}
				</span>
			</Link>
		)
	}
}
