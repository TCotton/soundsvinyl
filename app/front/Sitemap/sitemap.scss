@import '../../assets/sass/tools';

.sitemap {
	margin-bottom: 50px;
}

.sitemapUnorderedList {
	list-style: none;
	margin-bottom: 30px;
	li {
		display : block;
	}
}

.sitemapItem {
	@include font-calculator($font_family_body, 14px, 0);
	color: $paragraphColour;
	line-height: 1.5em;
	display: inline-block;
}
