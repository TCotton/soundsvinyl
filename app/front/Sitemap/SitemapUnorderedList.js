/* eslint-disable security/detect-object-injection */
import React, { Component } from 'react'
import './sitemap.scss'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

export default class SitemapUnorderedList extends Component {
	static propTypes = {
		categories: PropTypes.bool,
		data: PropTypes.arrayOf(
			PropTypes.object,
			PropTypes.string,
			PropTypes.number,
			PropTypes.node,
			PropTypes.element,
			PropTypes.arrayOf(PropTypes.string, PropTypes.number)
		)
	}

	static defaultProps = {
		categories: false,
		data: []
	}

	constructor (props) {
		super(props)
	}

	render () {
		const { categories, data } = this.props

		return (
			<ul styleName='sitemapUnorderedList'>
				{!categories &&
					data.map(data => {
						return (
							<li key={data._id}>
								<Link styleName='sitemapItem' to={`/${data.slug}/${data._id}`}>
									{data.title}
								</Link>
							</li>
						)
					})}

				{categories &&
					data.map((newData, index) => {
						const uri = 'category/' + encodeURIComponent(data[index][0])
						return (
							<li key={index.toString()}>
								<Link styleName='sitemapItem' to={uri}>
									{data[index][0]}
								</Link>
							</li>
						)
					})}
			</ul>
		)
	}
}
