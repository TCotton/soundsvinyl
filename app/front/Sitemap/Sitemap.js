import React, { Component } from 'react'
import './sitemap.scss'
import axios from 'axios'
import { homeURI } from '../../helper_constants'
import SitemapUnorderedList from './SitemapUnorderedList'
import ErrorBoundary from '../errorBoundaries/ErrorBoundary'
import groupBy from 'lodash/groupBy'
import sortBy from 'lodash/sortBy'

export default class Sitemap extends Component {
	constructor (props) {
		super(props)

		this.state = {
			customPages: [],
			error: null,
			pages: []
		}
	}

	componentDidMount () {
		axios
			.get(`${homeURI}/apiV1/page/getall`)
			.then(res => {
				this.setState({
					pages: res.data
				})
			})
			.catch(error => {
				this.setState({
					error: error.toString()
				})
				new Error(error.toString())
			})

		axios
			.get(`${homeURI}/apiV1/page/getTags`)
			.then(res => {
				let anArray = []

				// sorting of categories takes place at the API level with the use of aggregate
				// and then here to place the categories in a readable, clickable order

				res.data.forEach(element => {
					anArray.push(element._id.name)
				})

				const group = groupBy(anArray)
				const pairs = sortBy(group)

				this.setState({
					customPages: pairs
				})
			})
			.catch(error => {
				this.setState({
					error: error.toString()
				})
				new Error(error.toString())
			})
	}

	render () {
		const { error, pages, customPages } = this.state
		const $sitemap = 'Sitemap'
		const $Categories = 'Categories'
		const $Tracks = 'Tracks'

		return (
			<main styleName='sitemap'>
				<h3>{$sitemap}</h3>
				<section>
					{error && <p>{error}</p>}
					{!error && <h4>{$Categories}</h4>}
					{!error && (
						<ErrorBoundary>
							<SitemapUnorderedList categories data={customPages} />
						</ErrorBoundary>
					)}
					{!error && <h4>{$Tracks}</h4>}
					{!error && (
						<ErrorBoundary>
							<SitemapUnorderedList data={pages} />
						</ErrorBoundary>
					)}
				</section>
			</main>
		)
	}
}
