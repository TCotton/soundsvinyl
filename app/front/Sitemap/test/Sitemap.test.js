import React from 'react'
import Sitemap from '../sitemap'
import renderer from 'react-test-renderer'
import { MemoryRouter as Router } from 'react-router-dom'
import PageUnit from '../../Page/PageUnit'

describe('Component', () => {
	let component
	beforeEach(() => {
		component = renderer.create(<Sitemap />)
	})

	describe('Sitemap', () => {
		it('should be defined', () => {
			expect(component).toBeDefined()
		})

		it( 'snapshot', () => {

			const sitemap = renderer
				.create(
					<Sitemap />
				)
				.toJSON();

			expect( sitemap ).toMatchSnapshot();
		});
	})
})
