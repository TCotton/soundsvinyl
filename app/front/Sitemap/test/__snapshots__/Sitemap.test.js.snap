// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Sitemap snapshot 1`] = `
<main
  className="sitemap__sitemap___2mkRz"
>
  <h3>
    Sitemap
  </h3>
  <section>
    <h4>
      Categories
    </h4>
    <ul
      className="sitemap__sitemapUnorderedList___14wVG"
    />
    <h4>
      Tracks
    </h4>
    <ul
      className="sitemap__sitemapUnorderedList___14wVG"
    />
  </section>
</main>
`;
